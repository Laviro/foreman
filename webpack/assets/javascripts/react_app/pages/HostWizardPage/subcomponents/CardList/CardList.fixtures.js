import logo from '../../assets/host_group.png';

export const items = [
  {
    title: 'Host Group 1',
    logo,
  },
  {
    title: 'Host Group 2',
    logo,
  },
  {
    title: 'Host Group 3',
    logo,
  },
  {
    title: 'Host Group 4',
    logo,
  },
  {
    title: 'Host Group 5',
    logo,
  },
  {
    title: 'Host Group 6',
    logo,
  },
];
