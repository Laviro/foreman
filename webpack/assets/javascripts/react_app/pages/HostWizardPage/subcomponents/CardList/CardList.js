import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'patternfly-react';
import Card from '../Card';

const CardList = ({ items }) => (
  <Grid>
    {items.map(item => (
      <Grid.Col xs={6} sm={3}>
        <Card {...item} />
      </Grid.Col>
    ))}
  </Grid>
);

CardList.propTypes = {
  items: PropTypes.array,
};

CardList.defaultProps = {
  items: [],
};

export default CardList;
