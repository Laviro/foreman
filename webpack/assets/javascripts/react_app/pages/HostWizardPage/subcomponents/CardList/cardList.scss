.wizard_card_list {
  .pagination_arrow {
    font-size: 40px;
    height: 720px;
    display: flex;
    justify-content: center;
    cursor: pointer;
    transition: 0.5s;

    &.disabled {
      color: #d8d8d8;

      &:hover {
        background: none;
        cursor: default;
      }
    }

    &:hover {
      background-color: #f6f6f6;
    }

    .fa {
      align-self: center;
    }
  }
}
