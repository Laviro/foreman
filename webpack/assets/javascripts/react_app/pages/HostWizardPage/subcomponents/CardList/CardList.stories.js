import React from 'react';
import { storiesOf } from '@storybook/react';
import CardList from './CardList';
import { items } from './CardList.fixtures';

storiesOf('Components/HostWizard/Card', module).add('Basic Card List', () => (
  <div style={{ padding: '20px' }}>
    <CardList items={items} />
  </div>
));
