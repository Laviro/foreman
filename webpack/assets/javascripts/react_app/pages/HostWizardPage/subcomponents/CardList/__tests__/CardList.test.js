import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import CardList from '../CardList';

const fixtures = {
  'render without Props': {},
  /** fixtures, props for the component */
};

describe('CardList', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(CardList, fixtures));
});
