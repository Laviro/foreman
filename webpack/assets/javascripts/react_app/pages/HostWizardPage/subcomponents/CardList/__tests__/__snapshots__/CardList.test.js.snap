// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList rendering render without Props 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={false}
/>
`;
